### PR TITLE
Add tests covering v0.38.* thresholds regressions

### LINKED ISSUE #2525: Add integration tests for thresholds
### Feature Description

`v0.38.0` outlined its set of issues (#2520, #2518, #2512) with the final streak of switching to a Go-based thresholds engine. Some of them could have been caught beforehand if we had had a set of integration tests outlining the expected behavior of thresholds in known corner cases.

Let's find out the best way to write integration tests for k6's thresholds feature. The goal would be to outline what is the expected behavior of k6's thresholds in all the scenarios we wish to support.

#### Threshold over sub-metrics without samples should not result in `NaN`
```javascript
import { check, sleep } from 'k6';

import http from 'k6/http';

export const options = {
  scenarios: {
    iWillFail: {
      exec: 'iWillFail',
      executor: 'constant-vus',
      startTime: '2s',
      vus: 1,
      duration: '30s',
    },
  },

  thresholds: {
    'checks{type:read}': [{ threshold: 'rate>0.9', abortOnFail: true }],
  },
};

export function iWillFail() {
  let res = http.get(`https://test-api.k6.io/`);

  check(res, {
    'read status is 200': (r) => r.status === 200,
  }, { type: 'read' });

  sleep(1);
}
```

### Already existing or connected issues / PRs (optional)

#2521

### PR DESCRIPTION
**Pull request by oleiade**
This Pull Request addresses #2525 and adds tests covering the thresholds-related regressions observed in `v0.38.*` versions as a result of further replacement of the thresholds' evaluation logic.

Specifically, this PR adds tests covering #2518, #2512 and #2520.

I've tried different approaches for testing, but adding test scripts illustrating the issue, and writing tests for the `run` command, just like with did for thresholds validation at init ended up being the simplest, clearest and most convenient.

closes #2525

### CODE REVIEW

**Review by na-- — APPROVED** [review]

**Comment by na--** [comment]
though you should probably change the PR description to "closes https://github.com/grafana/k6/issues/2525", so the issue will be closed automatically once the PR is merged

File: cmd/run_test.go [hunk 1]
```diff
@@ -283,3 +284,61 @@ func TestInvalidOptionsThresholdErrExitCode(t *testing.T) {
...
+		{
+			name:         "#2512: parsing threshold names containing parsable tokens should be valid",
+			testFilename: "thresholds/name_contains_tokens.js",
+			expExitCode:  0,
+		},
+		{
+			name:                 "#2520: thresholds over metrics without values should avoid division by zero and displaying NaN values",
+			testFilename:         "thresholds/empty_sink_no_nan.js",
+			expExitCode:          0,
+			expStdoutNotContains: "NaN",
```
Comment (mstoykov) [inline cmd/run_test.go:314]: can we also check that it contains the rate but with a 0?

**Review by mstoykov — APPROVED** [review]

**Review by olegbespalov — DISMISSED** [review]

**Review by na-- — APPROVED** [review]

**Review by mstoykov — APPROVED** [review]
